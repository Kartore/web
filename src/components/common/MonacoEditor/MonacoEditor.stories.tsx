import type { Meta, StoryObj } from '@storybook/react';

import { MonacoEditor } from '.';

const meta = {
	component: MonacoEditor,
} satisfies Meta<typeof MonacoEditor>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
	args: {},
};
