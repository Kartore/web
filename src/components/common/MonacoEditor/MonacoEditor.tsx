import type { FC } from 'react';

import type { EditorProps } from '@monaco-editor/react';
import { Editor, loader } from '@monaco-editor/react';
import { shikiToMonaco } from '@shikijs/monaco';
import * as monaco from 'monaco-editor';
// eslint-disable-next-line import-x/default
import editorWorker from 'monaco-editor/esm/vs/editor/editor.worker?worker';
// eslint-disable-next-line import-x/default
import jsonWorker from 'monaco-editor/esm/vs/language/json/json.worker?worker';
import { createHighlighter } from 'shiki/bundle/web';

self.MonacoEnvironment = {
	getWorker(_, label) {
		if (label === 'json') {
			return new jsonWorker();
		}
		return new editorWorker();
	},
};

loader.config({ monaco });

export type MonacoEditorProps = {} & EditorProps;

export const MonacoEditor: FC<MonacoEditorProps> = ({
	options,
	onMount,
	...props
}) => {
	return (
		<Editor
			defaultLanguage={'json'}
			theme={'github-light'}
			options={{
				minimap: { enabled: false },
				scrollBeyondLastLine: false,
				...options,
			}}
			{...props}
			onMount={(editor, monaco) => {
				onMount?.(editor, monaco);
				(async () => {
					const highlighter = await createHighlighter({
						themes: ['github-light'],
						langs: ['json'],
					});
					monaco.languages.register({ id: 'json' });
					shikiToMonaco(highlighter, monaco);
				})();
			}}
		/>
	);
};

MonacoEditor.displayName = 'MonacoEditor';
