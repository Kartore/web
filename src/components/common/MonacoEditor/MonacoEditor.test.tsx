import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './MonacoEditor.stories';

const { Default } = composeStories(Stories);

describe('Component: MonacoEditor', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
