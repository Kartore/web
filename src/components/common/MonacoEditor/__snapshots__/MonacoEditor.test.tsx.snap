// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: MonacoEditor > Snapshot > Default 1`] = `
<DocumentFragment>
  <section
    style="display: flex; position: relative; text-align: initial; width: 100%; height: 100%;"
  >
    <div
      style="display: flex; height: 100%; width: 100%; justify-content: center; align-items: center;"
    >
      Loading...
    </div>
    <div
      style="width: 100%; display: none;"
    />
  </section>
</DocumentFragment>
`;
